src/librbd/ImageWatcher.cc

```diff
@@ -844,7 +844,6 @@ void ImageWatcher::handle_payload(const FlattenPayload &payload,
 
 void ImageWatcher::handle_payload(const ResizePayload &payload,
 				  bufferlist *out) {
-  RWLock::RLocker l(m_image_ctx.owner_lock);
```

Comment:
Right now, all operations are canceled / cleaned up / flushed when ownership is lost.  In reality, those checks in AsyncXyxRequest to see if ownership was lost should actually be asserts.  If you don't hold the lock, this cleanup logic would no longer function properly.